pkg/vfs/vfs.go

```diff
@@ -448,17 +463,35 @@ func (v *VFS) Truncate(ctx Context, ino Ino, size int64, opened uint8, attr *Att
 		err = syscall.EFBIG
 		return
 	}
-	hs := v.findAllHandles(ino)
-	sort.Slice(hs, func(i, j int) bool { return hs[i].fh < hs[j].fh })
-	for _, h := range hs {
+	if fh == 0 {
+		hs := v.findAllHandles(ino)
+		sort.Slice(hs, func(i, j int) bool { return hs[i].fh < hs[j].fh })
+		for _, h := range hs {
+			if !h.Wlock(ctx) {
+				err = syscall.EINTR
+				return
+			}
+			defer func(h *handle) { h.Wunlock() }(h)
+		}
+		_ = v.writer.Flush(ctx, ino)
+		err = v.Meta.Truncate(ctx, ino, 0, uint64(size), attr, false)
+	} else {
+		h := v.findHandle(ino, fh)
+		if h == nil {
+			err = syscall.EBADF
+			return
+		}
+		if h.writer == nil {
+			err = syscall.EACCES
+		}
 		if !h.Wlock(ctx) {
 			err = syscall.EINTR
 			return
 		}
 		defer func(h *handle) { h.Wunlock() }(h)
+		_ = h.writer.Flush(ctx)
```

Comment:
Why not flush all the fd?

Reply:
We could leave this one out of current PR

Reply:
fixed